pkg/sql/sem/tree/expr.go

```diff
@@ -884,7 +884,7 @@ func (node *Array) Format(ctx *FmtCtx) {
 	// If the array has a type, add an annotation. Don't add it if the type is
 	// UNKNOWN[], since that's not a valid annotation.
 	if ctx.HasFlags(FmtParsable) && node.typ != nil {
-		if node.typ.ArrayContents().Family() != types.UnknownFamily {
+		if node.typ.ArrayContents().Family() != types.UnknownFamily && node.typeAnnotation.typ == nil {
```

Comment:
to summarize what we discussed offline: `node.typ` and `node.typeAnnotation.typ` refer to the same thing, so this change doesn't have the intended affect.

perhaps instead our approach should be to update `formatViewQueryForDisplay` to detect when an annotation would be duplicated, and prevent that.